backend/dal/PIMSContext.cs

```diff
@@ -103,6 +109,31 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             base.OnModelCreating(modelBuilder);
         }
 
+        public override int SaveChanges()
+        {
+            // get entries that are being Added or Updated
+            var modifiedEntries = ChangeTracker.Entries()
+                    .Where(x => (x.State == EntityState.Added || x.State == EntityState.Modified));
+            var userId = _httpContextAccessor.HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value;
+            if (userId != null)
+            {
+                foreach (var entry in modifiedEntries)
+                {
+                    var entity = entry.Entity as BaseEntity;
+
+                    if (entry.State == EntityState.Added)
+                    {
+                        entity.CreatedById = new Guid(userId);
+                        entity.CreatedOn = DateTime.UtcNow;
+                    }
+                    entity.UpdatedById = new Guid(userId);
```

Comment:
Don't do this if it's a `EntityState.Delete` as it will most likely confuse EF with related properties.

Reply:
sure, updating.